distributed/http/tests/test_worker_http.py

```diff
@@ -35,3 +50,26 @@ def test_services(s, a, b):
     assert isinstance(c.services['http'], HTTPServer)
     assert c.service_ports['http'] == c.services['http'].port
     assert s.worker_services[c.address]['http'] == c.service_ports['http']
+
+
+@gen_cluster()
+def test_with_data(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield tornado.gen.sleep(0.5)
+    yield e._start()
+    yield e._scatter([1])
+    yield tornado.gen.sleep(0.5)
+    client = AsyncHTTPClient()
+    d = yield e.scheduler.identity()
+    sch_hport = services['http']
+    (wip, wport) = list(e.workers)[0]
```

Comment:
It is the tricky thing of getting the HTTP address out that's stumping me. Getting the scheduler's alone would be enough, and query that for the workers, or getting the worker ones directly would be fine too. The new `identity()` was supposed to solve that, but the port number doesn't come through in test mode.